jujugui/static/gui/src/app/views/topology/service.js

```diff
@@ -57,6 +63,17 @@ YUI.add('juju-topology-service', function(Y) {
     this.node = vis.selectAll('.service')
                      .data(Y.Object.values(topo.service_boxes),
                            function(d) {return d.modelId;});
+    // It takes a few update cycles to add the services to the database so
+    // this checks to make sure we only center once we have some services to
+    // center on.
+    if (this.centerOnLoad && serviceCount > 0) {
+      this.centerOnLoad = false;
+      // Bounce this panToCenter to the top of the stack so that the async
+      // d3 renderings complete before we try to pan.
+      setTimeout(() => {
```

Comment:
Hopefully this doesn't cause us issues in the future with the rendering order, something to bear in mind if we see any oddness.